### PR TITLE
SNOW-1445838: Support DataFrame.items.

### PR DESCRIPTION
**Pull request by sfc-gh-mvashishtha**
Fixes SNOW-1445838

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/dataframe.py [hunk 1]
```diff
@@ -1348,27 +1348,18 @@ def iterrow_builder(s):
...
+
         def items_builder(s):
             """Return tuple of the given `s` parameter name and the parameter themselves."""
             return s.name, s
 
         partition_iterator = PartitionIterator(self, 1, items_builder)
         yield from partition_iterator
 
-    @dataframe_not_implemented()
-    def iteritems(self):  # noqa: RT01, D200
```
Comment (sfc-gh-mvashishtha) [inline src/snowflake/snowpark/modin/pandas/dataframe.py:1365]: pandas 2.0 [removed](https://pandas.pydata.org/docs/whatsnew/v2.0.0.html#removal-of-prior-version-deprecations-changes) `iteritems`, an alias for `items`, in pandas 2.0, so no need to include it here.

File: tests/unit/modin/test_unsupported.py [hunk 2]
```diff
@@ -135,20 +135,6 @@ def test_unsupported_df(df_method, kwargs):
         getattr(mock_df, df_method)(**kwargs)
 
 
-@pytest.mark.parametrize(
-    "df_method, kwargs",
-    [["items", {}], ["iteritems", {}]],
-)
-def test_unsupported_df_generator(df_method, kwargs):
```
Comment (sfc-gh-mvashishtha) [inline tests/unit/modin/test_unsupported.py:142]: - pandas 2.0 [removed](https://pandas.pydata.org/docs/whatsnew/v2.0.0.html#removal-of-prior-version-deprecations-changes) `iteritems`, an alias for `items`, in pandas 2.0, so no need to include it here.
- We now support `items`.

**Review by sfc-gh-jkew — APPROVED** [review]
LGTM!

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM. Thanks, Mahesh!

**Pull request merged**